domapping/cli.py

```diff
@@ -73,6 +74,13 @@ def schema_to_mapping_cli(schema, output, config, indent):
             config_instance.load(json.load(conf))
 
     mapping = schema_to_mapping(parsed_schema, id, {}, config_instance)
+    if final:
+        mapping_name = os.path.splitext(os.path.basename(schema.name))[0]
```

Comment:
the file name is not reliable. If the schema is coming from `stdin` there is no `schema.name`. This name should be given as an argument to the option `estype`.